src/main/scala/com/github/unchama/seichiassist/subsystems/tradesystems/subsystems/gachatrade/bukkit/listeners/GachaTradeListener.scala

```diff
@@ -0,0 +1,76 @@
+package com.github.unchama.seichiassist.subsystems.tradesystems.subsystems.gachatrade.bukkit.listeners
+
+import cats.effect.ConcurrentEffect.ops.toAllConcurrentEffectOps
+import cats.effect.{ConcurrentEffect, IO}
+import com.github.unchama.seichiassist.concurrent.PluginExecutionContexts.onMainThread
+import com.github.unchama.seichiassist.subsystems.gacha.GachaAPI
+import com.github.unchama.seichiassist.subsystems.gacha.domain.CanBeSignedAsGachaPrize
+import com.github.unchama.seichiassist.subsystems.tradesystems.subsystems.gachatrade.bukkit.traderules.BukkitTrade
+import com.github.unchama.seichiassist.util.InventoryOperations
+import com.github.unchama.util.InventoryUtil.InventoryOps
+import org.bukkit.ChatColor._
+import org.bukkit.Sound
+import org.bukkit.entity.Player
+import org.bukkit.event.inventory.InventoryCloseEvent
+import org.bukkit.event.{EventHandler, Listener}
+import org.bukkit.inventory.ItemStack
+
+class GachaTradeListener[F[_]: ConcurrentEffect](
+  implicit canBeSignedAsGachaPrize: CanBeSignedAsGachaPrize[ItemStack],
```

Comment:
`domain` で
```
trait GachaListProvider[F[_], ItemStack] {
  readGachaList: F[Vector[GachaPrize[ItemStack]]]
}

trait GachaTradeRule[ItemStack] {
  def ruleFor(playerName: String, gachaList: Vector[GachaPrize[ItemStack]]) => TradeRule[ItemStack]
}
```
として、
```suggestion
class GachaTradeListener[F[_]: ConcurrentEffect](
  rule: GachaTradeRule[ItemStack]
)(
  implicit canBeSignedAsGachaPrize: CanBeSignedAsGachaPrize[ItemStack],
  gachaListProvider: GachaListProvider[F, ItemStack],
```
とすると、リスナーと `BukkitTradeRule` の間にインターフェースを張れるので、こうしたほうがいいかも